src/api/directives.ts

```diff
@@ -74,4 +79,27 @@ const submitDirective = async (
     }
 };
 
-export { exchangeBearerToken, submitDirective };
+const getApplieDirectives = (type: keyof typeof DIRECTIVES, userId: string) => {
+    let queryBuilder = supabaseClient.from<JoinedAppliedDirective>(
+        TABLES.APPLIED_DIRECTIVES
+    ).select(`
+            id,
+            directive_id,
+            job_status,
+            logs_token,
+            user_id,
+            user_claims,
+            updated_at,
+            directives !inner(spec->>type)
+        `);
+
+    queryBuilder = queryBuilder.eq('directives.spec->>type', type);
+    queryBuilder = queryBuilder.eq('user_id', userId);
+
+    return DIRECTIVES[type]
+        .queryFilter(queryBuilder)
+        .order('updated_at', { ascending: false })
+        .limit(1);
```

Comment:
I have been wondering if queries that should only return a single result should be explicitly limited. How do you feel about imposing explicit limits where expected?

Reply:
I have been wondering about that. I think it is a good idea when we know the exact count since if you do this and there are multiples returned then an error is thrown. 

However, I kinda of don't like queries with limits ever and would rather just always have arrays and handle multiples from day one (in case we ever need to handle multiples).

In our system I am feeling we're probably safe limiting the queries we think should because they are really rare so kinda of nice to call out those out.